File: arrays/find.pivot.index.ts
/**
 * @param {number[]} nums
 * @return {number}
 */
export function pivotIndex(nums) {
  let leftSum = 0;
  let rightSum = nums.reduce((a, b) => a + b) - nums[0];

  for (let i = 0; i < nums.length; i++) {
    if (leftSum === rightSum) {
      return i;
    }

    leftSum += nums[i];
    if (i + 1 < nums.length) {
      rightSum -= nums[i + 1];
    }
  }

  return -1;
}

File: arrays/find_pivot_index.go
package arrays

func pivotIndex(nums []int) int {
	leftSum := 0
	rightSum := 0

	for _, v := range nums[1:] {
		rightSum += v
	}

	for i := 0; i < len(nums); i++ {
		if leftSum == rightSum {
			return i
		}

		leftSum += nums[i]

		if i+1 < len(nums) {
			rightSum -= nums[i+1]
		}
	}

	return -1
}
